### PR TITLE
suppress display of -x and -N on argparse help

### LINKED ISSUE #1232: Can we remove -x and -N options from -h and doc/user/scripts.txt output?
With the new -M and -U we don't need them any more and I think we would prefer they be hidden.

ping @mr-c

### PR DESCRIPTION
**Pull request by ctb**
Fixes #1232.

### CODE REVIEW

**Comment by ctb** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

**Comment by ctb** [comment]
Ready for review & merge, @mr-c. I think this is the final planned code change, yes?

**Comment by ctb** [comment]
retest this please, jenkins.

**Comment by mr-c** [comment]
Re: https://github.com/dib-lab/khmer/issues/1232#issuecomment-128987924 I'd rather not supress these options at this time.

Maybe amend them to remind people that there is an easier way?

**Comment by mr-c** [comment]
Bumping to 2.0+

**Comment by standage** [comment]
@ctb So to be clear, the idea is to hide `-x/-N` for typical user but have some kind of verbose help that includes a description of the "power" options?

**Comment by ctb** [comment]
On Mon, Jun 27, 2016 at 08:10:41PM -0700, Daniel Standage wrote:
> @ctb So to be clear, the idea is to hide `-x/-N` for typical user but have some kind of verbose help that includes a description of the "power" options?

Yep.

**Comment by betatim** [comment]
Refreshed the PR. It looked like this was basically ready to go and then got forgotten?

**Comment by ctb** [comment]
We left it here: https://github.com/dib-lab/khmer/pull/1235#issuecomment-129261058 - we didn't like the SUPPRESS approach. Alternatives?

**Comment by betatim** [comment]
Ah, that is a fail on reading comprehension for me :)

What you want is to hide it by default for `blah.py --help` but show it when you do `blah.py --HELP` (or what ever)? Off the top of my head the way to do this is to peek at `argv` and detect that the flag for "really help me this time" is present and swap from suppressing to showing the help message. Pretty hacky.

> Maybe amend them to remind people that there is an easier way?

Seems much easier/less likely to break.

**Comment by ctb** [comment]
On Thu, Nov 17, 2016 at 07:01:27AM -0800, Tim Head wrote:
> Ah, that is a fail on reading comprehension for me :)
> 
> What you want is to hide it by default for `blah.py --help` but show it when you do `blah.py --HELP` (or what ever)? Off the top of my head the way to do this is to peek at `argv` and detect that the flag for "really help me this time" is present and swap from suppressing to showing the help message. Pretty hacky.
> 
> > Maybe amend them to remind people that there is an easier way?
> 
> Seems much easier/less likely to break.

or,

* have the default output (when run with no args) not show them;
* have --help show them under "advanced arguments"

or something like that.

You COULD do something like use args.SUPPRESS, and then add an argument
--help-expert, which shows all the arguments you're not supposed to use
except in case of emergency... hmm, I like that.

**Comment by betatim** [comment]
Added `--help-expert`, somewhat hacky. Ideas how to make it nicer? Otherwise ready for review @ctb

**Comment by standage** [comment]
I think this should go into 2.1.

**Comment by betatim** [comment]
Ready for review! @luizirber, @camillescott, @standage , or @ctb

File: tests/test_script_arguments.py [hunk 1]
```diff
@@ -122,6 +122,34 @@ def test_check_tablespace_nodegraph(graph_type, exp_buckets):
     assert sizestr == exp_buckets
 
 
+def test_normal_help(capsys):
+    # check -x and -N are hiden by default with --help
```
Comment (standage) [inline tests/test_script_arguments.py]: hiden --> hidden

**Comment by betatim** [comment]
Shall we wait for @ctb to hit the button then?

**Comment by standage** [comment]
Nope, just had to wait for the build to complete after my typo fix. Merging now.

**Pull request merged**